pkg/controller/restic_repository_controller.go

```diff
@@ -285,37 +253,11 @@ func repoReady() func(*velerov1api.ResticRepository) {
 // patchResticRepository mutates req with the provided mutate function, and patches it
 // through the Kube API. After executing this function, req will be updated with both
 // the mutation and the results of the Patch() API call.
-func (c *resticRepositoryController) patchResticRepository(req *velerov1api.ResticRepository, mutate func(*velerov1api.ResticRepository)) error {
-	// Record original json
-	oldData, err := json.Marshal(req)
-	if err != nil {
-		return errors.Wrap(err, "error marshalling original ResticRepository")
-	}
-
+func (r *resticRepoConciler) patchResticRepository(ctx context.Context, req *velerov1api.ResticRepository, patchHelper *patch.Helper, log logrus.FieldLogger, mutate func(*velerov1api.ResticRepository)) error {
```

Comment:
Cannot use `patchHelper` in this way.

When you call `patchHelper.Patch(newObj)`, it compares the `oldObj` with `newObj`, the `oldObj` is the one passed when creating the helper. So if you use the same helper to patch different `newObjs`, you may get unexpected results

Reply:
because it needs to patch the same cr object status many times in one function(reconcile), so I used one patchHelper for the same cr object. PodVolumeRestoreReconciler also has the same usage